shell/platform/common/text_editing_delta.cc

```diff
@@ -0,0 +1,37 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/common/text_editing_delta.h"
+
+namespace flutter {
+
+TextEditingDelta::~TextEditingDelta() = default;
+
+TextEditingDelta::TextEditingDelta(std::u16string textBeforeChange,
+                                   TextRange range,
+                                   std::u16string text) {
+  int start = range.start();
+  int end = range.start() + range.length();
+  setDeltas(textBeforeChange, text, start, end);
```

Comment:
Here and below: I'd be tempted to get rid of `SetDeltas` altogether and just move initialisation of the instance variables to the initialiser list:

```cc
TextEditingDelta::TextEditingDelta(std::u16string textBeforeChange,
                                   TextRange range,
                                   std::u16string text) 
  : oldText_(textBeforeChange),
    deltaText_(text),
    deltaStart_(range.start()),
    deltaEnd_(range.start() + range.length()) {
} 
```
Note that the order in the initialiser list _must_ be the same as the order of declaration in the struct.

Reply:
I was able to get rid of SetDeltas, thanks.